lib/active_merchant/billing/gateways/pay_gate_xml.rb

```diff
@@ -201,6 +208,10 @@ def build_request(action, options={})
               money           = options.delete(:money)
               authorization   = options.delete(:authorization)
               build_capture(protocol, money, authorization, options)
+            when 'refundtx'
+              money           = options.delete(:money)
+              authorization   = options.delete(:authorization)
+              build_refund(protocol, money, authorization, options)
```

Comment:
The block
```ruby 
money = options.delete(:money)
authorization = options.delete(:authorization)
``` 
is repeated unnecessarily inside each case block. I'd suggest you take the opportunity to clean this up, i.e. move these lines above the `case` statement such that it looks like: 
```ruby 
when 'authtx'
  build_authorization(...) 
when 'settletx' 
  build_capture(...)
when 'refundtx' 
  build_refund(...) 
```